xbmc/windowing/osx/WinSystemOSX.mm

```diff
@@ -243,6 +281,55 @@ CFArrayRef GetAllDisplayModes(CGDirectDisplayID display)
   return displayModes;
 }
 
+CGDisplayModeRef GetModeById(std::string& modeId, NSUInteger screenIdx)
+{
+  if (modeId.empty())
+    return nullptr;
+
+  bool stretched;
+  bool interlaced;
+  bool safeForHardware;
+  size_t resWidth;
+  size_t resHeight;
+  size_t pixelWidth;
+  size_t pixelHeight;
+  size_t bitsperpixel;
+  RESOLUTION_INFO res;
+
+  CLog::LogF(LOGDEBUG, "Looking for mode for screen {} with id {}", screenIdx, modeId);
+
+  CFArrayRef allModes = GetAllDisplayModes(GetDisplayID(screenIdx));
```

Comment:
Well, I am not really familiar with this stuff. I assume it's just a matter of wrapping into an `@autoreleasepool` and avoid the calls to `CFRelease`?
Please check last patch revision

Reply:
unfortunately, `@autoreleasepool` can't help us because it's from objc / Foundation world, and here we're in CoreFoundation one (which is pure C btw). You'd rather rename the function to `CreateAllDisplayModes` (or CopyAllDisplayModes) and also CFRelease the returned value at the calling site (but don't forget to check for nullptr), as it's obtained with CGDisplay**Copy**AllDisplayModes.

if you absolutely want to use autoreleased objects, then you must return them from function as `CFAutorelease(obj)`.